nepalingo-web/src/App.tsx

```diff
@@ -1,12 +1,36 @@
 import React from "react";
+import {
+  BrowserRouter as Router,
+  Route,
+  Routes,
+  Navigate,
+} from "react-router-dom";
+import User_auth from "./components/user_auth/User_auth";
+import Home from "./pages/Home/Home";
 import Header from "./components/Header";
+import supabase from "./components/user_auth/supabaseClient";
+
+const App: React.FC = () => {
+  const [isAuthenticated, setIsAuthenticated] = React.useState(false);
+
+  React.useEffect(() => {
+    const session = supabase.auth.getSession();
+    setIsAuthenticated(!!session);
```

Comment:
I think this returns a promise?

Reply:
Use async/await here.

Example: https://supabase.com/docs/guides/auth/quickstarts/react

Reply:
done